dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelAIPluginExtensions.cs

```diff
@@ -0,0 +1,406 @@
+using Microsoft.SemanticKernel.SkillDefinition;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+using Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Extensions;
+
+public static class KernelAIPluginExtensions
+{
+    /// <summary>
+    /// Imports an AI plugin that is exposed as an OpenAPI v3 endpoint or through OpenAI's ChatGPT format.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="filePath">The file path to the AI Plugin</param>
+    /// <param name="executionParameters">Skill execution parameters.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns></returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportAIPluginAsync(
+        this IKernel kernel,
+        string skillName,
+        string filePath,
+        OpenApiSkillExecutionParameters? executionParameters = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(kernel);
+        Verify.ValidSkillName(skillName);
+
+#pragma warning disable CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
+        var internalHttpClient = HttpClientProvider.GetHttpClient(kernel.Config, executionParameters?.HttpClient, kernel.Log);
+#pragma warning restore CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
+
+        var pluginContents = await LoadDocumentFromFilePath(
+            kernel,
+            filePath,
+            executionParameters,
+            internalHttpClient,
+            cancellationToken).ConfigureAwait(false);
+
+        return await CompleteImport(
+            kernel,
+            pluginContents,
+            skillName,
+            internalHttpClient,
+            executionParameters,
+            cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Imports an AI plugin that is exposed as an OpenAPI v3 endpoint or through OpenAI's ChatGPT format.
+    /// </summary>
+    /// <param name="kernel">Semantic Kernel instance.</param>
+    /// <param name="skillName">Skill name.</param>
+    /// <param name="uri">A local or remote URI referencing the AI Plugin</param>
+    /// <param name="executionParameters">Skill execution parameters.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns></returns>
+    public static async Task<IDictionary<string, ISKFunction>> ImportAIPluginAsync(
+        this IKernel kernel,
+        string skillName,
```

Comment:
Should skillName be a required arg? I would expect this can be grabbed from the "name_for_model" property in the plugin schema, if not provided by a user.

Reply:
Seems reasonable - what happens if a dev imports a plugin without a name (which we supply via name_for_model or maybe the Title if it's an OpenAPI endpoint we are loading), then later wants to reference the plugin by name? 

They would have the object reference given the current method signature, but not the assigned name unless we modify the signature

Reply:
Spoke offline, will revisit this in a separate issue.